puffin/src/lib.rs

```diff
@@ -447,15 +504,27 @@ pub struct GlobalProfiler {
 
     next_sink_id: FrameSinkId,
     sinks: std::collections::HashMap<FrameSinkId, FrameSink>,
+    propagate_all_scope_details: bool,
+
+    // When scope details are reported by the macros.
+    new_scopes: Vec<ScopeDetails>,
+    // When scope details are reported by external libraries.
+    new_custom_scopes: Vec<Arc<ScopeDetails>>,
```

Comment:
Can these be combined into one? Using `Arc` for both seems like a pretty small price to pay for simpler code (one extra allocation for each new scope)

EDIT: we do that `Arc::new` in new_frame anyways

Reply:
Yes, think was that one has the scope id set and the other not. but i can check if scope id is none instead.